kratos/python_scripts/python_solver.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+
+class PythonSolver(object):
+    """The base class for the Python Solvers in the applications
+    Changes to this BaseClass have to be discussed first!
+    """
+    def __init__(self, model_part, solver_settings):
+        """The constructor of the PythonSolver-Object.
+
+        It is intended to be called from the constructor
+        of deriving classes:
+        super(DerivedSolver, self).__init__(solver_settings)
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        model_part -- The ModelPart to be used
+        solver_settings -- The solver settings used
+        """
+        if (type(model_part) != KratosMultiphysics.ModelPart):
+            raise Exception("Input is expected to be provided as a Kratos Model object")
+
+        if (type(solver_settings) != KratosMultiphysics.Parameters):
+            raise Exception("Input is expected to be provided as a Kratos Parameters object")
+
+        self.main_model_part = model_part
+        self.solver_settings = solver_settings
+
+    def AddVariables(self):
+        pass
+
+    def AddDofs(self):
+        pass
+
+    def GetMinimumBufferSize(self):
+        pass
+
+    def ReadModelPart(self):
+        # TODO replace the functions in the solvers in Fluid and Structure
+        KratosMultiphysics.Logger.PrintInfo("::[PythonSolver]::", "Reading model part.")
```

Comment:
One thing I was wondering is if we should "hard-code" labels like this. If we call it from a derived solver, is it more useful for the label to be "[PythonSolver]" or "[DynamicStructuralSolver]"? I don't have a strong opinion on this, but it is a discussion I think we should have at some point.

Reply:
I thing, I remember or something... That in my analysis I added a method to return the label, that is overrided in each solver, so the solver always knows the name of the solver

Reply:
It would be easy to add a GetSolverName function that every solver implements

Might be useful if several solvers are used, which is likely to be the case in the future 

I dont habe a strong opinion though

Reply:
Another point for hardcoding is that if a solver calls the functions of the baseclass (e.g. `Initialize`) then we would have multiple `Initializing done` prints with the same solver name which I think would be quite confusing